vanguard.go

```diff
@@ -248,11 +274,30 @@ func NewService(servicePath string, handler http.Handler, opts ...ServiceOption)
 			err: fmt.Errorf("could not resolve schema for service at path %q: resolved descriptor is %s, not a service", servicePath, descKind(desc)),
 		}
 	}
-	return NewServiceWithSchema(svcDesc, handler, opts...)
+	return &Service{
```

Comment:
Not using the constructor seems to be a side affect of the option smuggling. I think we can still use it if the resolver initializer if moved away from an option.

Reply:
We'd still need to smuggle some bit to indicate this is a static service that can use global types, whereas the other is a potentially dynamic service that needs a resolver computed.

I guess we can just _always_ go through the logic for a dynamic service. It should recognize that it can use global types and then not bother computing a separate registry...